docs/cookbook/shared-state.md

```diff
@@ -0,0 +1,113 @@
+---
+title: Shared State
+description: How to use a reactive composable to share your objects across component files.
+author: /
+thumbnail: /recipes/animations.png
+difficulty: 0
+---
+
+# Shared State in TresJS
+
+This guide will help you get started with shared state in TresJS by building a simple scene with a cube that can be shared across component files.
+
+<StackBlitzEmbed project-id="tresjs-minimal-reproduction-rycc4j" />
+
+## Creating a State Composable
+
+First, we'll create a composable to store the objects.
+
+```ts
+//composables/state.ts
+import { reactive, toRefs } from "vue";
+
+const state = reactive({
+    mesh: null,
+    //you can add more objects here
+})
+
+export function useState() {
+    return {
+        ...toRefs(state)
+    }
+}
+```
+
+## Setting the object to the state
+
+Next, we'll assign an object to the state and include it in a subcomponent where we can access and use it.
+
+```vue
+//Parent component
```

Comment:
You can take advantage of [vitepress code groups](https://vitepress.dev/guide/markdown#code-groups ) to organize several code blocks on tabs with filenames


See an example of our documentation here 

https://github.com/Tresjs/tres/blob/8d53bbae37d858ddf775f0574647e86979473ac4/docs/cookbook/load-models.md?plain=1#L68-L101

Reply:
I have changed the blocks to use code groups. I also added a pinia example.

Reply:
Hi @whitespacecode there is a vue tab that is empty in the latest commit

![Screenshot 2024-11-05 at 15 36 38](https://github.com/user-attachments/assets/993ef9b6-2bda-4f78-b01b-140610b09e7e)

Reply:
This should be fixed now